applications/trilinos_application/external_includes/mpi_amgcl_ns_solver.h

```diff
@@ -0,0 +1,391 @@
+//#include "Teuchos_ParameterList.hpp"
+
+
+#include <amgcl/amg.hpp>
+#include <amgcl/adapter/epetra.hpp>
+#include <amgcl/profiler.hpp>
+
+#include <amgcl/mpi/make_solver.hpp>
+#include <amgcl/mpi/schur_pressure_correction.hpp>
+#include <amgcl/mpi/block_preconditioner.hpp>
+#include <amgcl/mpi/subdomain_deflation.hpp>
+#include <amgcl/mpi/direct_solver.hpp>
+#include <boost/property_tree/json_parser.hpp> //needed to print AMGCL internal settings
+
+
+
+namespace Kratos
+{
+
+
+
+template< class TSparseSpaceType, class TDenseSpaceType,
+          class TReordererType = Reorderer<TSparseSpaceType, TDenseSpaceType> >
+class AmgclMPI_NS_Solver : public LinearSolver< TSparseSpaceType,
+    TDenseSpaceType, TReordererType>
+{
+public:
+    /**
+     * Counted pointer of AmgclMPI_NS_Solver
+     */
+    KRATOS_CLASS_POINTER_DEFINITION ( AmgclMPI_NS_Solver );
+
+    typedef LinearSolver<TSparseSpaceType, TDenseSpaceType, TReordererType> BaseType;
+
+    typedef typename TSparseSpaceType::MatrixType SparseMatrixType;
+
+    typedef typename TSparseSpaceType::VectorType VectorType;
+
+    typedef typename TDenseSpaceType::MatrixType DenseMatrixType;
+
+
+    AmgclMPI_NS_Solver ( Parameters rParameters )
+    {
+//         Parameters default_parameters( R"(
+//                                        {
+//                                        "solver_type" : "AmgclMPI_NS_Solver",
+//                                        "krylov_type" : "fgmres",
+//                                        "velocity_block_preconditioner" :
+//                                         {
+//                                             "krylov_type" : "lgmres",
+//                                             "tolerance" : 1e-3,
+//                                             "preconditioner_type" : "ilu0",
+//                                             "max_iteration": 5
+//                                         },
+//                                         "pressure_block_preconditioner" :
+//                                         {
+//                                             "krylov_type" : "lgmres",
+//                                             "tolerance" : 1e-2,
+//                                             "preconditioner_type" : "spai0",
+//                                             "max_iteration": 20
+//                                         },
+//                                        "tolerance" : 1e-9,
+//                                        "gmres_krylov_space_dimension": 50,
+//                                        "coarsening_type": "aggregation",
+//                                        "max_iteration": 50,
+//                                        "verbosity" : 1,
+//                                        "scaling": false,
+//                                        "coarse_enough" : 5000
+//                                    }  )" );
+// 
+// 
+//         //now validate agains defaults -- this also ensures no type mismatch
```

Comment:
makes sense. Actually it was a mistake to commit this in this branch, but at this point i'll fix it here